lib/src/rules/avoid_init_to_null.dart

```diff
@@ -66,21 +68,30 @@ class AvoidInitToNull extends LintRule implements NodeLintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    final visitor = _Visitor(this);
+    final visitor = _Visitor(this, context);
     registry.addVariableDeclaration(this, visitor);
     registry.addDefaultFormalParameter(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
   final LintRule rule;
+  final LinterContext context;
 
-  _Visitor(this.rule);
+  final bool nnbdEnabled;
+  _Visitor(this.rule, this.context)
+      : nnbdEnabled = context.analysisOptions.contextFeatures
```

Comment:
This is not fully correct.
Individual libraries might specify their language versions, so `FeatureSet`.
The feature set from the context is the default value when not overridden on the library level.

So, ask the unit for its `CompilationUnit.featureSet` instead.

Reply:
excellent.  Paul pointed this out too.  Thank you!